sfcasts/ep1/mocking-stubs.md

```diff
@@ -1,52 +1,55 @@
-bend PHP unit, and awesome. Our test pass. The best thing about this is we're no
-longer calling GitHubs API, ever retirement, run our tests. Imagine what that would
-do in a, a continuous integration environment, such as getting up actions, where if
-you're running this test, thousands of times a day, it's calling that API over and
-over again. Remember when we were talking about, uh, all the different names for
-mock, well, both a response or a mock response, and our mock client here are, uh, are
-called stubs. Stubs are fake objects where you optionally take control of its return
-values, which we're doing in this case, when we're saying will return for both of
-these objects. Again, the terminology isn't too important, but there you go. These
-are stubs. And yes, every time I teach this, I need to look up these terms to
-remember exactly what they mean
-
-Coming up next. We're going to take our coming up next. We're going to return our
-stubs into mock objects, which is basically the same thing, but we're also going to,
-uh, control the data that we are passing into these, to the two array and the request
-method.
-
-Yeah.
+Let's take a quick look back at our `GithubService` and see exactly what we're
+doing in here. First, we're passing in an `HttpClientInterface` into the service,
+then we make a HTTP Request, and we ultimately expect an `HttpResponseInterface`
+object with our GitHub data. We transform that data into an array, iterating
+over each item within that array to get the health status of a specific dino using
+its name.
 
+To get our tests to pass, we need to *teach* our fake `HttpClient` that when
+we call the `request()` method, it should return a `Response` object containing
+data that we control. So let's do that...
+
+Right after the `$mockHttpClient`, add `$mockResponse = $this->createMock()` using
+`ResponseInterface::class` for the name. Below on `$mockHttpClient`, call 
+`method('request')` and `willReturn($mockResponse)`. This tells our mock client
+that hey, anytime we call the request method on our mock, you need to return *this*
+mock response.
+
+We *could* our tests now, but they would fail. We taught our mock http client
+*what* it should return when we call the `request()` method. *But*, we need
+to teach our `$mockResponse` what *it* needs to do when we call the `toArray()`
+method. So right above, lets teach the `$mockResponse` that when we call the
+`method('toArray')`, it `willReturn()` an `[]` of issues. As that's what GitHub
+returns when we call the API.
+
+For each issue, GitHub gives us, among other things, the issue's "title" along with
+an array of "labels" for that issue. So let's mimic GitHub and add an issue `[]`
+that has the `'title' => 'Daisy'`. Then for the `'labels' => []`, add a label `[]`
+with `name => Status: Sick`.
+
+To be realistic, copy this issue and paste it below. Change `Daisy` to `Maverick`.
+And because he has a test flight at noon, he has the `Status: Healthy` label.
```

Comment:
I'm getting hung up here.. Specifically

>So let's mimic GitHub and add an issue `[]` that has the `'title' => 'Daisy'`.

Our code is an array within an array that has arrays within arrays... haha

```
willReturn([
    [
        'title' -> 'Daisy',
        'labels' => [ ['name' => 'Status: Sick'] ],
......
```

Reply:
How about?

```diff
-So let's mimic GitHub and add an issue [] that has the 'title' => 'Daisy'.
+So let's mimic GitHub and make this array include one issue that has the 'title' => 'Daisy'.
```

```diff
- Then for the `'labels' => []`, add a label `[]` with `name => Status: Sick`.
+ Also give this issue some `labels` set to an array that includes one: `name => Status: Sick`
```

Overall, I thought what you had was fine though. I do sometimes say and write `[]` (read as left square bracket right square bracket usually), but I don't think it's needed in this case.